### DISCUSSION TITLE
Can `BlockDiagLinearOperator` subsume `BlockInterleavedLinearOperator`?

### DISCUSSION THREAD

**Discussion by SebastianAment** [cornellius-gp/linear_operator#16] [Ideas]
I noticed that the doc strings of [`BlockDiagLinearOperator`](https://github.com/cornellius-gp/linear_operator/blob/a58efe780c90ff8fa19c85aa1dfee6f5b61c6373/linear_operator/operators/block_diag_linear_operator.py#L35) and [`BlockInterleavedLinearOperator`](https://github.com/cornellius-gp/linear_operator/blob/main/linear_operator/operators/block_interleaved_linear_operator.py), as well as their implementations, are mostly identical (for the subset of methods implemented by the latter). Also, the only use of the interleaved operator that I found is [here](https://github.com/cornellius-gp/gpytorch/blob/fd2de04fa09a469eaccc6339dd8c275d3e950d2c/gpytorch/distributions/multitask_multivariate_normal.py#L115). 

Is there a current or planned difference in functionality between the two, or could the interleaved operator be subsumed by the block diagonal one? If possible, this would reduce code complexity without compromising features, further benefitting test coverage and maintainability.

**Comment by Balandat** [comment]
Yeah good point. The name suggests that `BlockInterleavedLinearOperator` was introduced to deal with "interleaved' representations of multi-output MVNs, but at least a cursory pass through the code doesn't really reveal why we need this. At the very least it seems that one should be able to combine the two classes as you said.
